docs/ldml/tr35-keyboards.md

```diff
@@ -1369,7 +1382,7 @@ _Attribute:_ `modifiers` (required for `hardware`)
 1. `L` or `R` indicates a left- or right- side modifier only (such as `altL`)
  whereas `alt` indicates _either_ left or right alt key (that is, `altL` or `altR`). `ctrl` indicates either left or right ctrl key (that is, `ctrlL` or `ctrlR`).
 
-2.  If there are any layers (in the same `form=`) with a modifier `alt`, there may not also be another layer with `altL` or `altR`.  Similarly, if there is a layer with a modifier `ctrl`, there may not be a layer with `ctrlL` or `ctrlR`.
+2. Keyboard implementations must warn keyboard authors if a keyboard mixes `alt` with `altL`/`altR`, or `ctrl` with `ctrlL`/`ctrlR`.
```

Comment:
```suggestion
2. Mixing `alt` with `altL`/`altR`, or `ctrl` with `ctrlL`/`ctrlR`, is strongly discouraged. Keyboard implementations detect and should warn keyboard authors if this occurs.
```

We can't say "must" if there might not be a UI for alerting authors.

Reply:
As this is an interchange format, it's assumed that there could be a UI or similar (such as a build step which could fail).

Reply:
The phrasing is just weird. I author a keyboard, send it to you, you run a program on it. How does the program know who I am to notify me? Maybe "must emit a detect and emit a warning"?

Reply:
changed.